xbmc/games/controllers/types/PortNode.cpp

```diff
@@ -70,6 +70,20 @@ CControllerNode& CPortNode::GetActiveController()
   return invalid;
 }
 
+bool CPortNode::SetActiveController(const std::string& controllerId)
+{
+  for (unsigned int i = 0; i < m_controllers.size(); ++i)
+  {
+    if (m_controllers.at(i).GetController()->ID() == controllerId)
```

Comment:
Is the pointer here guaranteed to be valid?

Reply:
It doesn't make sense for an empty pointer to end up in an array of objects, but I added a null check just in case.

Reply:
I think there are a few of these but I'm fine with your explanation 😊

Reply:
I found two other occurrences and added defensive checks